server/src/main/java/org/opensearch/index/query/QueryShardContext.java

```diff
@@ -522,6 +535,70 @@ private ParsedQuery toQuery(QueryBuilder queryBuilder, CheckedFunction<QueryBuil
         }
     }
 
+    public ParsedQuery toStarTreeQuery(
+        CompositeIndexFieldInfo starTree,
+        CompositeDataCubeFieldType compositeIndexFieldInfo,
+        QueryBuilder queryBuilder,
+        Query query
+    ) {
+        Map<String, List<StarTreeFilter.Range>> queryMap;
+
+        if (queryBuilder == null) {
+            queryMap = null;
+        } else if (queryBuilder instanceof TermQueryBuilder) {
```

Comment:
So while debugging, I checked that `query` is actually an instance of `IndexOrDocValueQuery` - which could have ben rewritten from queries other than a term query. So to keep my checks limited and the scope of influence under control - I checked the instance of its builder instead. For `IndexOrDocValueQuery` - I might have to further check other criteria as well.

Reply:
It's too bad that `org.opensearch.search.aggregations.bucket.filterrewrite.Helper#unwrapIntoConcreteQuery` isn't in a reusable place.

It could live in `org.opensearch.common.lucene.search.Queries`.